lib/cmake/libcudacxx/libcudacxx-config.cmake

```diff
@@ -0,0 +1,56 @@
+#
+# find_package(libcudacxx) config file.
+#
+# Defines a libcudacxx::libcudacxx target that may be linked from user projects to include
+# libcudacxx.
+
+if (TARGET libcudacxx::libcudacxx)
+  return()
+endif()
+
+function(_libcudacxx_declare_interface_alias alias_name ugly_name)
+  # 1) Only IMPORTED and ALIAS targets can be placed in a namespace.
+  # 2) When an IMPORTED library is linked to another target, its include
+  #    directories are treated as SYSTEM includes.
+  # 3) nvcc will automatically check the CUDA Toolkit include path *before* the
+  #    system includes. This means that the Toolkit libcudacxx will *always* be used
+  #    during compilation, and the include paths of an IMPORTED libcudacxx::libcudacxx
+  #    target will never have any effect.
+  # 4) This behavior can be fixed by setting the property NO_SYSTEM_FROM_IMPORTED
+  #    on EVERY target that links to libcudacxx::libcudacxx. This would be a burden and a
+  #    footgun for our users. Forgetting this would silently pull in the wrong libcudacxx!
+  # 5) A workaround is to make a non-IMPORTED library outside of the namespace,
+  #    configure it, and then ALIAS it into the namespace (or ALIAS and then
+  #    configure, that seems to work too).
+  add_library(${ugly_name} INTERFACE)
```

Comment:
This approach won't work when the consumer is part of an install or build export set.
Adding the following to the `test_export` exposes the issue:

```cmake
# Install tree exporting of version_check_objs
install(TARGETS version_check_objs EXPORT test-targets)
install(EXPORT test-targets FILE test-targets.cmake DESTINATION "lib/cmake/libcudacxx-test/")

# Build tree exporting of version_check_objs
export(EXPORT test-targets FILE "${CMAKE_CURRENT_BINARY_DIR}/libcudacxx-test/test-targets.cmake")
```

The reason this approach works for Thrust is that every project has to construct a <namespace>::Thrust target 
via `thrust_create_target` that is IMPORTED INTERFACE. I believe that what we can do is just make ` alias_name` the firewall and things should work for the above use-cases.

```cmake
  add_library(${ugly_name} INTERFACE)

  add_library(${alias_name} INTERFACE IMPORTED GLOBAL)
  target_link_libraries(${alias_name} INTERFACE  ${ugly_name})
```

Reply:
Makes sense -- fixed in last push.